### PR TITLE
NamedExec Bulk Insert Fix

### LINKED ISSUE #657: NamedExec/slice does not work with Postgres INSERT ... ON CONFLICT
First of all, thanks for this fantastic lib, Jason! It has saved me countless hours in implementing database interfaces!

I recently came across a new(?) feature which allows a NamedExec to receive a slice of structs. This works beautifully with a simple INSERT statement, generating a `VALUES (...), (...), (....) ...` list.

However, it does not work with the Postgres `INSERT INTO table VALUES (...) ON CONFLICT DO UPDATE SET column1 = :gofield1, column2 = :gofield2, ...` type of statement, as the arguments for the latter part are not "multiplied out" for all slice elements, but only appended once to the args array.

### PR DESCRIPTION
**Pull request by abraithwaite**
This change-set is how I addressed the issues of bulk inserts not being handled cleanly when they're not at the end of the query.

It works by first adding `VALUES` to the expression (not sure if this is universally applicable to all supported SQL dialects) then adding a submatch to the regular expression for the fields we want to repeat.  Using the submatch, we're able to repeat the fields that we're concerned with to expand the insertion parameters.

Likely? Fixes #657, #694, #705, #709, #712, #713 ?

### CODE REVIEW

File: named.go [hunk 1]
```diff
@@ -224,21 +224,28 @@ func bindStruct(bindType int, query string, arg interface{}, m *reflectx.Mapper)
...
-	if len(loc) != 2 {
+	loc := valueBracketReg.FindAllStringSubmatchIndex(bound, -1)
+	// Either no VALUES () found or more than one found??
+	if len(loc) != 1 {
+		return bound
+	}
+	// defensive guard. loc should be len 4 representing the starting and
+	// ending index for the whole regex match and the starting + ending
+	// index for the single inside group
+	if len(loc[0]) != 4 {
```
Comment (abraithwaite) [inline named.go:238]: If `loc[0] != 4`, that indicates that the regex was changed and either the submatcher was removed or another submatcher was added.

**Comment by gsora** [comment]
Just tried this patch myself and it works fine with `INSERT ... ON CONFLICT DO UPDATE SET`.

File: named_test.go [hunk 2]
```diff
@@ -214,7 +218,7 @@ func TestNamedQueries(t *testing.T) {
 		}
 
 		_, err = db.NamedExec(`INSERT INTO person (first_name, last_name, email)
-			VALUES (:first_name, :last_name, :email) `, slsMap)
+			VALUES (:first_name, :last_name, :email) ;--`, slsMap)
```
Comment (jmoiron) [inline named_test.go:221]: Is this extra comment necessary?

Comment (abraithwaite) [reply named_test.go:221]: The comment here is exercising the regex in the sense that anything after the closing paren is ignored.

I could update it to something more clear like `ON CONFLICT` perhaps?  I imagine that most people are using that behavior moreso than random junk at the end.

Comment (jmoiron) [reply named_test.go:221]: Happy with the comment, just did not want to add some extra load bearing requirement here.  I think it's clear from the tests later on that these are testing some boundary conditions.

**Comment by jmoiron** [comment]
Tests looked good, tagged as v1.3.2
